tests/loader_version_tests.cpp

```diff
@@ -315,3 +315,93 @@ TEST(MultipleDriverConfig, DifferentICDInterfaceVersions) {
+    ASSERT_EQ(phys_dev_count, 4);
+}
+
+TEST(MultipleDriverConfig, DifferentICDsWithDevicesAndGroups) {
+    MultipleICDShim env({TestICDDetails(TEST_ICD_PATH_EXPORT_ICD_GIPA), TestICDDetails(TEST_ICD_PATH_VERSION_2),
+                         TestICDDetails(TEST_ICD_PATH_VERSION_2_EXPORT_ICD_GPDPA)},
+                        DebugMode::none);
+
+    // The loader has to be able to handle drivers that support device groups in combination
+    // with drivers that don't support device groups.  When this is the case, the loader needs
+    // to take every driver that doesn't support device groups and put each of its devices in
+    // a separate group.  Then it combines that information with the drivers that support
+    // device groups returned info.
+
+    // ICD 0 :  No 1.1 support (so 1 device will become 1 group in loader)
+    TestICD& icd0 = env.get_test_icd(0);
+    icd0.physical_devices.emplace_back("physical_device_0");
+    icd0.min_icd_interface_version = 5;
+    icd0.max_icd_interface_version = 5;
+    icd0.SetICDAPIVersion(VK_API_VERSION_1_0);
+
+    // ICD 1 :  1.1 support (with 1 group with 2 devices)
+    TestICD& icd1 = env.get_test_icd(1);
+    icd1.physical_devices.emplace_back("physical_device_1");
+    icd1.physical_devices.emplace_back("physical_device_2");
+    icd1.physical_device_groups.emplace_back(icd1.physical_devices[0]);
+    icd1.physical_device_groups.back().use_physical_device(icd1.physical_devices[1]);
+    icd1.min_icd_interface_version = 5;
+    icd1.max_icd_interface_version = 5;
+    icd1.SetICDAPIVersion(VK_API_VERSION_1_1);
+
+    // ICD 2 :  No 1.1 support (so 3 devices will become 3 groups in loader)
+    TestICD& icd2 = env.get_test_icd(2);
+    icd2.physical_devices.emplace_back("physical_device_3");
+    icd2.physical_devices.emplace_back("physical_device_4");
+    icd2.physical_devices.emplace_back("physical_device_5");
+    icd2.min_icd_interface_version = 5;
+    icd2.max_icd_interface_version = 5;
+    icd2.SetICDAPIVersion(VK_API_VERSION_1_0);
+
+    InstWrapper inst{env.vulkan_functions};
+    inst.create_info.set_api_version(1, 1, 0);
+    inst.CheckCreate();
+
+    uint32_t group_count = static_cast<uint32_t>(5);
+    uint32_t returned_group_count = 0;
+    ASSERT_EQ(VK_SUCCESS, inst->vkEnumeratePhysicalDeviceGroups(inst, &returned_group_count, nullptr));
+    ASSERT_EQ(group_count, returned_group_count);
+
+    std::vector<VkPhysicalDeviceGroupProperties> group_props{};
+    group_props.resize(group_count);
+    group_props[0].sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_GROUP_PROPERTIES;
+    group_props[1].sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_GROUP_PROPERTIES;
+    group_props[2].sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_GROUP_PROPERTIES;
+    group_props[3].sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_GROUP_PROPERTIES;
+    group_props[4].sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_GROUP_PROPERTIES;
```

Comment:
```suggestion
    group_props.resize(group_count, VkPhysicalDeviceGroupProperties{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_GROUP_PROPERTIES});
```
Resize accepts a default value that is used, makes the code more DRY.